test/csuite/incr_backup/main.c

```diff
@@ -0,0 +1,846 @@
+        }
+        free(value);
+        free(value2);
+        testutil_check(cur->close(cur));
+    }
+}
+
+/*
+ * create_table --
+ *     Create a table for the given slot.
+ */
+static void
+create_table(WT_SESSION *session, TABLE_INFO *tinfo, uint32_t slot)
+{
+    char *uri;
+
+    testutil_assert(!TABLE_VALID(&tinfo->table[slot]));
+    uri = dcalloc(1, URI_MAX_LEN);
+    testutil_check(
+      __wt_snprintf(uri, URI_MAX_LEN, URI_FORMAT, (int)slot, (int)tinfo->table[slot].name_index++));
+
+    VERBOSE(3, "create %s\n", uri);
+    testutil_check(session->create(session, uri, "key_format=S,value_format=u"));
+    tinfo->table[slot].name = uri;
+    tinfo->tables_in_use++;
+}
+
+static void
+rename_table(WT_SESSION *session, TABLE_INFO *tinfo, uint32_t slot)
+{
+    char *olduri, *uri;
+
+    testutil_assert(TABLE_VALID(&tinfo->table[slot]));
+    uri = dcalloc(1, URI_MAX_LEN);
+    testutil_check(
+      __wt_snprintf(uri, URI_MAX_LEN, URI_FORMAT, (int)slot, (int)tinfo->table[slot].name_index++));
+
+    olduri = tinfo->table[slot].name;
+    VERBOSE(3, "rename %s %s\n", olduri, uri);
+    testutil_check(session->rename(session, olduri, uri, NULL));
+    free(olduri);
+    tinfo->table[slot].name = uri;
+}
+
+static void
+drop_table(WT_SESSION *session, TABLE_INFO *tinfo, uint32_t slot)
+{
+    char *uri;
+
+    testutil_assert(TABLE_VALID(&tinfo->table[slot]));
+    uri = tinfo->table[slot].name;
+
+    VERBOSE(3, "create %s\n", uri);
+    testutil_check(session->drop(session, uri, NULL));
+    free(uri);
+    tinfo->table[slot].name = NULL;
+    tinfo->table[slot].change_count = 0;
+    tinfo->tables_in_use--;
+}
+
+static void
+check_table(WT_SESSION *session, TABLE *table)
+{
+    WT_CURSOR *cursor;
+    WT_ITEM item, got_value;
+    OPERATION_TYPE op_type;
+    uint64_t boundary, change_count, expect_records, got_records, total_changes;
+    int keylow, keyhigh, ret;
+    u_char *value;
+    char *got_key;
+    char key[MAX_KEY_SIZE];
+
+    expect_records = 0;
+    total_changes = table->change_count;
+    boundary = total_changes % 10000;
+    op_type = (OPERATION_TYPE)(total_changes % 40000) / 10000;
+    value = dcalloc(1, table->max_value_size);
+
+    VERBOSE(3, "Checking: %s\n", table->name);
+    switch (op_type) {
+    case INSERT:
+        expect_records = total_changes % 10000;
+        break;
+    case UPDATE:
+    case MODIFY:
+        expect_records = 10000;
+        break;
+    case REMOVE:
+        expect_records = 10000 - (total_changes % 10000);
+        break;
+    }
+
+    testutil_check(session->open_cursor(session, table->name, NULL, NULL, &cursor));
+    got_records = 0;
+    while ((ret = cursor->next(cursor)) == 0) {
+        got_records++;
+        testutil_check(cursor->get_key(cursor, &got_key));
+        testutil_check(cursor->get_value(cursor, &got_value));
+        testutil_assert(sscanf(got_key, KEY_FORMAT, &keylow, &keyhigh) == 2);
+        change_count = (u_int)keyhigh * 100 + (u_int)keylow;
```

Comment:
A brief comment reminding us about how we're doing this checking here would be helpful.